### ISSUE TITLE
LastPass extension not working with FF 71.

### ISSUE THREAD

**Issue by pbogdan** [NixOS/nixpkgs#75818]
**Describe the bug**

Unable to authenticate in the LastPass extension.

**To Reproduce**

1. Install Firefox.
2. Install the LastPass extension.
3. Attempt to authenticate:
- when invoking the extension via its toolbar icon simply nothing happens
- when invoking it through a pop up dialog when prompted to supply credentials to an account stored in LastPass the screen gets stuck on the spinner icon

**Expected behavior**

LastPass is functional.

**Additional context**

Digging a bit into it lead me to https://bugzilla.mozilla.org/show_bug.cgi?id=1601707 however it appears some of form of the patch has been incorporated both into master and 19.09 (https://github.com/NixOS/nixpkgs/pull/75169 & https://github.com/NixOS/nixpkgs/pull/75184). As far as I can tell however it wasn't the final version of the patch from the upstream bug report.
A rebuild with the final version applied to an earlier 19.09 fixed my particular issue. I can't however vouch for any impact on other extensions nor am I presently able to re-test atop latest 19.09.
`firefox-bin` package works without any issues.

**Metadata**

- system: `"x86_64-linux"`
 - host os: `Linux 5.4.3, NixOS, 19.09.1594.fbe321e6669 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(root): `"nixos-19.09.1594.fbe321e6669"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: firefox{,-unwrapped}
# a list of nixos modules affected by the problem
module:
```

**Comment by pbogdan** [comment]
Swapped [the patch](https://github.com/NixOS/nixpkgs/blob/release-19.09/pkgs/applications/networking/browsers/firefox/fix-ff71-lto.patch) for [the one from the upstream bug tracker](https://phabricator.services.mozilla.com/D56876) and this appears to have fixed my particular issue. Logging in / out using the extension works and so does filling in credentials. Have not tested anything beyond this single extension.

/cc @fpletz @andir @Ma27 in case they can offer any feedback on whether the patch currently included may need to reviewed / revisited.

**Comment by Ma27** [comment]
IIRC the patch which was available back when this issue was investigated didn't fix all the problems in FF we found, so unless they updated the patch we probably need a combination of both patches, but @fpletz probably knows more details :)

**Comment by pbogdan** [comment]
Hate to bump this again - while not impossible to workaround this issue is getting somewhat annoying in day-to-day usage. Is there at least some way to establish provenance of the second patch so that I may try to combine them? I may have missed something obvious but I can only track down the one from the upstream issue I linked.

**Comment by Ma27** [comment]
Did you test this with a firefox that was built from source (including the upstream fix)?

**Comment by pbogdan** [comment]
I have indeed - built Firefox from this branch https://github.com/pbogdan/nixpkgs/tree/ff71-patch-udpate which at the time was based on top of latest release-19.09. The relevant commit is fc982fc5b0c963ed8b9bd16c7eccf427a661b13e. It's a diff of a diff which makes things confusing so to clarify I simply swapped the existing `fix-ff71-lto.patch` with https://bugzilla.mozilla.org/show_bug.cgi?id=1601707 -> CTRL-F -> pascalc : approval-mozilla-release-

At the very least it makes LastPass usable for me; I have that build still sitting around so I would be willing to check / test other extensions or issues that were encountered previously.

**Comment by obadz** [comment]
Well done tracking this down @pbogdan. I've also fallen victim to this bug and I'm now using `firefox-bin` to workaround. :+1: to any permanent solution.

**Comment by andir** [comment]
Can any of you test the folloing nixpkgs patch to see if that fixes all the observed issues?

```diff
diff --git a/pkgs/applications/networking/browsers/firefox/common.nix b/pkgs/applications/networking/browsers/firefox/common.nix
index a6d57ae8edd..2cd55a41d58 100644
--- a/pkgs/applications/networking/browsers/firefox/common.nix
+++ b/pkgs/applications/networking/browsers/firefox/common.nix
@@ -116,7 +116,13 @@ let
       url = "https://raw.githubusercontent.com/archlinuxarm/PKGBUILDs/09c7fa0dc1d87922e3b464c0fa084df1227fca79/extra/firefox/build-arm-libopus.patch";
       sha256 = "1zg56v3lc346fkzcjjx21vjip2s9hb2xw4pvza1dsfdnhsnzppfp";
     })
-  ] ++ lib.optional (lib.versionAtLeast ffversion "71") ./fix-ff71-lto.patch
+  ]
+  #++ lib.optional (lib.versionAtLeast ffversion "71") ./fix-ff71-lto.patch
+  ++ lib.optional (lib.versionAtLeast ffversion "71") (fetchpatch {
+      name = "fix-temporary-lifetimes.patch";
+      url = https://phabricator.services.mozilla.com/D56873?id=206659&download=true;
+      sha256 = "02dwd8mqx9rbm2d36ac2w1chmycygmcd5g1plbgbj577ghkv1fj6";
+    })
   ++ patches;
 
   nss_pkg = if lib.versionAtLeast ffversion "71" then nss_3_47_1 else nss;
```

**Comment by 9999years** [comment]
Also having this issue. @andir how do I test the patch? I'm a bit new to NixOS but I'd love to help.

**Comment by andir** [comment]
@9999years just use the latest release on the channels. It should carry all the required patches.

**Comment by pbogdan** [comment]
This is indeed fixed for me on latest 19.09. Is anyone else able to confirm as well before I go ahead and close this?
